core/modules/css/CssAccess.cc

```diff
@@ -0,0 +1,848 @@
+    auto paramMap = _getSubkeys(tableKey, subKeys);
+    if (paramMap.empty()) {
+        // check table key
+        if (not _kvI->exists(tableKey)) throw NoSuchTable(dbName, tableName);
+        return params;
+    }
+
+    // fill the structure
+    params.dirDb = paramMap["partitioning/dirDb"];
+    params.dirTable = paramMap["partitioning/dirTable"];
+    params.dirColName = paramMap["partitioning/dirColName"];
+    params.latColName = paramMap["partitioning/latColName"];
+    params.lonColName = paramMap["partitioning/lonColName"];
+    params.partitioned =
+            paramMap.count("partitioning") + paramMap.count("partitioning.json") > 0;
+    try {
+        auto iter = paramMap.find("partitioning/subChunks");
+        if (iter != paramMap.end()) {
+            params.subChunks = std::stoi(iter->second);
+        }
+        iter = paramMap.find("partitioning/overlap");
+        if (iter != paramMap.end()) {
+            params.overlap = std::stod(iter->second);
+        }
+    } catch (std::exception const& exc) {
+        throw KeyValueError(tableKey + "/partitioning", "one of the sub-keys is not numeric: " +
+                            std::string(exc.what()));
+    }
+
+    return params;
+}
+
+TableParams
+CssAccess::getTableParams(std::string const& dbName, std::string const& tableName) const {
+    LOGF(_log, LOG_LVL_DEBUG, "getTableParams(%1%, %2%)" % dbName % tableName);
+
+    std::string const tableKey = _prefix + "/DBS/" + dbName + "/TABLES/" + tableName;
+
+    TableParams params;
+
+    std::vector<std::string> subKeys{"partitioning/subChunks", "partitioning/dirDb",
+        "partitioning/dirTable", "partitioning/dirColName", "partitioning/latColName",
+        "partitioning/lonColName", "partitioning/overlap", "partitioning/secIndexColName",
+        "match/dirTable1", "match/dirColName1", "match/dirTable2", "match/dirColName2",
+        "match/flagColName", "partitioning", "partitioning.json"};
+    auto paramMap = _getSubkeys(tableKey, subKeys);
+    if (paramMap.empty()) {
+        // check table key
+        if (not _kvI->exists(tableKey)) throw NoSuchTable(dbName, tableName);
+        return params;
+    }
+
+    // fill the structure
+    params.match.dirTable1 = paramMap["match/dirTable1"];
+    params.match.dirColName1 = paramMap["match/dirColName1"];
+    params.match.dirTable2 = paramMap["match/dirTable2"];
+    params.match.dirColName2 = paramMap["match/dirColName2"];
+    params.match.flagColName = paramMap["match/flagColName"];
+    params.partitioning.dirDb = paramMap["partitioning/dirDb"];
+    params.partitioning.dirTable = paramMap["partitioning/dirTable"];
+    params.partitioning.dirColName = paramMap["partitioning/dirColName"];
+    params.partitioning.latColName = paramMap["partitioning/latColName"];
+    params.partitioning.lonColName = paramMap["partitioning/lonColName"];
+    params.partitioning.partitioned =
+            paramMap.count("partitioning") + paramMap.count("partitioning.json") > 0;
```

Comment:
It seems like maybe the subkeys and all the params.x = paramMap["x"] should be part of TableParams. With a getSubkeys() function and a loadFromMap(&map) function. It looks like there's a lot in common between MatchTableParams, TableParams, and the other Params structures, which could be leveraged to get rid of some nearly duplicate code. Might be better to do it in another ticket though.